src/parser/ParsedQuery.cpp

```diff
@@ -277,6 +278,10 @@ void ParsedQuery::GraphPattern::addLanguageFilter(const Variable& variable,
               QLEVER_INTERNAL_PREFIX_IRI_WITHOUT_CLOSING_BRACKET)) {
         matchingTriples.push_back(&triple);
       }
+      if (triple.s_ == variable || triple.o_ == variable ||
+          (isVariable(triple.p_) && triple.p_.iri_ == variable.name())) {
+        variableFoundInTriple = true;
+      }
```

Comment:
Add a comment, that if the variable appears in some other triple we can still join with the `ql:langag` predicate.